test/envtest/konnect_entities_target_test.go

```diff
@@ -0,0 +1,135 @@
+		updateKongUpstreamStatusWithProgrammed(t, ctx, clientNamespaced, upstream, upstreamID, cp.GetKonnectID())
+
+		t.Log("Setting up a watch for KongTarget events")
+		w := setupWatch[configurationv1alpha1.KongTargetList](t, ctx, cl, client.InNamespace(ns.Name))
+
+		t.Log("Setting up SDK expectations on Target creation")
+		sdk.TargetsSDK.EXPECT().CreateTargetWithUpstream(
+			mock.Anything,
+			mock.MatchedBy(func(req sdkkonnectops.CreateTargetWithUpstreamRequest) bool {
+				return *req.TargetWithoutParents.Target == targetHost && *req.TargetWithoutParents.Weight == int64(targetWeight)
+			}),
+		).Return(&sdkkonnectops.CreateTargetWithUpstreamResponse{
+			Target: &sdkkonnectcomp.Target{
+				ID: lo.ToPtr(targetID),
+			},
+		}, nil)
+
+		t.Log("Creating a KongTarget")
+		createdTarget := deploy.KongTargetAttachedToUpstream(t, ctx, clientNamespaced, upstream,
+			func(obj client.Object) {
+				kt := obj.(*configurationv1alpha1.KongTarget)
+				kt.Spec.KongTargetAPISpec.Target = targetHost
+				kt.Spec.KongTargetAPISpec.Weight = targetWeight
+			},
+		)
+		t.Log("Checking SDK KongTarget operations")
+		require.EventuallyWithT(t, func(c *assert.CollectT) {
+			assert.True(c, factory.SDK.TargetsSDK.AssertExpectations(t))
+		}, waitTime, tickTime)
+
+		t.Log("Waiting for Target to be programmed and get Konnect ID")
+		watchFor(t, ctx, w, watch.Modified, func(kt *configurationv1alpha1.KongTarget) bool {
+			return kt.GetKonnectID() == targetID && lo.ContainsBy(kt.Status.Conditions,
+				func(c metav1.Condition) bool {
+					return c.Type == "Programmed" && c.Status == metav1.ConditionTrue
+				})
+		}, "KongTarget didn't get Programmed status condition or didn't get the correct (target-12345) Konnect ID assigned")
+
+		t.Log("Setting up SDK expectations on Target update")
+		sdk.TargetsSDK.EXPECT().UpsertTargetWithUpstream(
+			mock.Anything,
+			mock.MatchedBy(func(req sdkkonnectops.UpsertTargetWithUpstreamRequest) bool {
+				return req.TargetID == targetID && *req.TargetWithoutParents.Weight == int64(200)
+			}),
+		).Return(&sdkkonnectops.UpsertTargetWithUpstreamResponse{}, nil)
+
+		t.Log("Patching KongTarget")
+		targetToPatch := createdTarget.DeepCopy()
+		targetToPatch.Spec.Weight = 200
+		require.NoError(t, clientNamespaced.Patch(ctx, targetToPatch, client.MergeFrom(createdTarget)))
+
+		t.Log("Waiting for Target to be updated in the SDK")
+		assert.EventuallyWithT(t, func(c *assert.CollectT) {
+			assert.True(c, factory.SDK.TargetsSDK.AssertExpectations(t))
+		}, waitTime, tickTime)
+
+		t.Log("Setting up SDK expectations on Target deletion")
+		sdk.TargetsSDK.EXPECT().DeleteTargetWithUpstream(
+			mock.Anything,
+			mock.MatchedBy(func(req sdkkonnectops.DeleteTargetWithUpstreamRequest) bool {
+				return req.TargetID == targetID
+			}),
+		).Return(&sdkkonnectops.DeleteTargetWithUpstreamResponse{}, nil)
+
+		t.Log("Deleting KongTarget")
+		require.NoError(t, clientNamespaced.Delete(ctx, createdTarget))
```

Comment:
Please add an eventual check that the target got deleted, otherwise we might go into checking the assertions which might have not fired just yet (i.e. the object did not get the `Delete()` called).

Reply:
Added.